controlplane/pkg/model/model.go

```diff
@@ -153,13 +179,13 @@ func (i *impl) SelectEndpoint(nsName string) (*registry.NSERegistration, error)
 	}
 	i.Lock()
 	defer i.Unlock()
-	idx := i.roundRobin[nsName] % len(nseRegistrations)
-	endpoint := nseRegistrations[idx]
-	if endpoint == nil {
-		return nil, errors.New("should not see this error, scaffolding called")
+
+	if len(*nsLabels) > 0 {
```

Comment:
If you pass in ns *registry.NetworkService ... then you can do this same len(ns.GetMatches()) test here.